### DISCUSSION TITLE
Sendfile support and configuration

### DISCUSSION THREAD

**Discussion by sevdog** [nginx/unit#1046] [Support] [answered]
Does nginx-unit supports [sendfile](https://www.nginx.com/resources/wiki/start/topics/examples/xsendfile/) like nginx does?

There is no reference about this topic in the current manual, while there are some source files which reference a "sendfile" mechanism.

**Comment by ac000** [comment]
AFAICT Unit will use an OS's sendfile(2) where it's available. This seems to be; Linux, FreeBSD, macOS, Solaris and HP-UX and AIX (not sure how much we really support those last two...).

I'm assuming this only works for static file sharing, can't say I've noticed any sendfile(2) system calls when strace(1)ing applications under Unit.

**Comment by sevdog** [comment]
What I would like to know if unit does recognize the `X-Accel-Redirect` header if provided by ther underlaying application, thuse serving directly any static resource without requiring the application to do it.

**Reply by ac000** [reply]
Ah OK, you meant ```XSendfile``` and not sendfile(2) the system call...

I see no mention of ```X-Accel-Redirect``` in the Unit source. I suspect the answer is to create a static share in Unit for such things.

**Reply by lcrilly** [reply, marked as answer]
A static share won't work on its own.

The typical use case is for the application logic to handle access control and if all is good then send the location of the resource in its response.

Instead of sending a 302 response with a Location header. The application can send a 200 response with the location in the `X-Accel-Redirect` header. This instructs the proxy to just send the resource directly.

Unit could potentially support this with [chaining](https://github.com/nginx/unit/issues/764).
